src/android/java/org/alohalytics/HttpTransport.java

```diff
@@ -66,27 +65,20 @@ public static Params run(final Params p) throws IOException, NullPointerExceptio
           throw new NullPointerException("Please set Content-Type for POST requests.");
         }
         connection.setRequestProperty("Content-Type", p.contentType);
+        if (p.contentEncoding != null) {
+          connection.setRequestProperty("Content-Encoding", p.contentEncoding);
+        }
         connection.setDoOutput(true);
         if (p.data != null) {
-          // Use gzip compression for memory-only transfers.
-          // TODO(AlexZ): Move compression to the lower file-level (file storage queue) to save device space.
-          final ByteArrayOutputStream bos = new ByteArrayOutputStream();
-          final GZIPOutputStream zos = new GZIPOutputStream(bos);
-          try {
-            zos.write(p.data);
-          } finally {
-            zos.close();
-          }
-          connection.setFixedLengthStreamingMode(bos.size());
-          connection.setRequestProperty("Content-Encoding", "gzip");
+          connection.setFixedLengthStreamingMode(p.data.length);
```

Comment:
Yes, it has moved into one place in C++ instead of many places in http impl.



> On 20.3.2015, at 03.48, Dima <notifications@github.com> wrote:
> 
> In src/android/java/org/alohalytics/HttpTransport.java:
> 
> >          connection.setDoOutput(true);
> >          if (p.data != null) {
> > -          // Use gzip compression for memory-only transfers.
> > -          // TODO(AlexZ): Move compression to the lower file-level (file storage queue) to save device space.
> > -          final ByteArrayOutputStream bos = new ByteArrayOutputStream();
> > -          final GZIPOutputStream zos = new GZIPOutputStream(bos);
> > -          try {
> > -            zos.write(p.data);
> > -          } finally {
> > -            zos.close();
> > -          }
> > -          connection.setFixedLengthStreamingMode(bos.size());
> > -          connection.setRequestProperty("Content-Encoding", "gzip");
> > +          connection.setFixedLengthStreamingMode(p.data.length);
> So, you remove gzip compression here?
> 
> —
> Reply to this email directly or view it on GitHub.
>